packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1943,6 +1943,8 @@ export class Bucket extends BucketBase {
     if (this.policy) {
       customResource.node.addDependency(this.policy);
     }
+
+    customResource.node.addDependency(this);
```

Comment:
There's already a CF dependency because the bucket name is referenced in the properties of the CR.

https://github.com/aws/aws-cdk/blob/492d33b27bc5b935e3da75f0bddd875bb6f9c15d/packages/%40aws-cdk/aws-s3/test/bucket.test.ts#L2437-L2439

Isn't the issue in #16332 some kind of race condition between the last published access log (coming from the delete of the main bucket) and the access log bucket deletion?

Reply:
Yes, I think you're right. Closing this PR.